### DISCUSSION TITLE
AWS.EKS  onhour/offhour filtering

### DISCUSSION THREAD

**Discussion by gabfelp** [cloud-custodian/cloud-custodian#7433] [General]
### Ask your question

Hi everyone and thanks for this tool. 
Cloud Custodian has been very helpful so far.

To explain my story: I created a policy that took care of my aws.EC2, aws.RDS and aws.ASG type instances with no issues, but I had some difficulties with the aws.EKS type.

What I'm trying to do is set an "offhour policy" for my aws.EKS clusters (or nodes) to stop/suspend/anything else that makes them unusable other than deleting.

My difficulty is that:
1 - the offhour/onhour filter does not seem to be set for aws.EKS (eks and eks-nodegroup) types. i.e. I cannot set specific hours to take actions.
2 - I don't know what kind of action I can put to do what I want, because "stop" and "suspend" are not possible either on EKS.

Is there any workaround for this?
Thanks!

### Policy

```shell
policies:
  - name: stop-after-hours-eks
    resource: eks
    filters:
      - type: offhour
        tag: Custodian
        opt-out: true
        default_tz: pt
        weekends: true # leave resources off for the weekend
        offhour: 15
    actions:
      - stop
```

### Relevant log/traceback output

```shell
custodian.commands:ERROR invalid policy file: aws_scheduler_policy.yml error: Failed to validate policy 'stop' is not of type 'object'
```

**Comment by kapilt** [comment]
eks doesn't really have a stop action,  you'd have to manage the underlying node groups and compute providers, and in some cases that requires reaching into the cluster services to configure their capacity. its not something easily managed from outside of the group. if you have cluster auto scaling setup, you can get a dueling control plane setup, where managing down on the compute side, can cause the k8s reconcillation groups to try and reschedule workloads onto new compute. additionally there is the possiblility of fargate compute or self managed nodes. tldr, you really want to handle this in cooperation with k8s control plane rather then external to k8s/eks. additionally 'stopping' eks is supported by aws, its simply the control plane for the eks cluster/api endpoints, not the workloads, and k8s workloads/nodes expect a control plane always active.


maybe it would be better to start with your goal? ie. you want to turn off node groups and workloads/services?

**Comment by gabfelp** [comment]
Thanks for the answer, @kapilt . 

I will revisit these possibilities in the future, but I ended up stuck trying to do a [similar thing](https://github.com/cloud-custodian/cloud-custodian/issues/7067)* with ECS-services for now. Stopping ECS-services became more relevant than stopping EKS at the moment for me. I wrote [another issue](https://github.com/cloud-custodian/cloud-custodian/issues/7067) about what I was experiencing while following the ECS documentation.

Thanks for your help so far!
